### DISCUSSION TITLE
[RFC] FloatButton support internal Badge

### DISCUSSION THREAD

**Discussion by MadCcc** [ant-design/ant-design#40861] [RFCs - archive]
## Summary

FloatButton support `badge` prop to add Badge to itself.

## Activation

After introducing FloatButton into antd in 5.0, we have received several issues about incorrect display of Badge with FloatButton (.e.g https://github.com/ant-design/ant-design/issues/39193). Since Badge depends on a element whose position is `absolute`, FloatButton cannot be used with it right now because of its `fixed` position.

Finally we decided to place Badge into FloatButton as internal feature. In this way, we could also control the placement of Badge with FloatButton to make it looks more naturally.

## API

| Property | Description | Type | Default | 
| --- | --- | --- | --- | 
| badge | Show Badge component. | BadgeProps | - | 

> Notice that Badge.Ribbon is not supported.

## Basic Example

```tsx
import React from 'react';
import { FloatButton } from 'antd';

export default () => (
  <FloatButton badge={{ count: 5 }} />
)
```

Then we would get a FloatButton like this:

![image](https://user-images.githubusercontent.com/27722486/220375294-31b01f7c-b335-487c-bdd5-6dc271330ebb.png)

### Placement

The offset is `6px` by default. Since Badge support `offset` to customize, it can also be other number if user want.

![image](https://user-images.githubusercontent.com/27722486/220376014-26e286b1-ff97-4a59-9b2d-49dd8bb72d90.png)

### Dot Badge

Dot Badge should always be placed on the border of FloatButton by Default, so the default offset should be calculated dynamicly (with Design Token).

![image](https://user-images.githubusercontent.com/27722486/220377068-46a08225-3664-4eff-a779-3400cdca84aa.png)

## How we teach this

In **feature** branch:

1. Make FloatButton support `badge` prop.
2. Provide demo and docs about `badge`.

## Full Design

![image](https://user-images.githubusercontent.com/27722486/220377447-e6e2ff48-13fb-4522-a725-75bff5bb979d.png)

**Comment by li-jia-nan** [comment]
mark

**Comment by afc163** [comment]
color 呢
